torch/quantization/fx/quantization_patterns.py

```diff
@@ -836,50 +877,72 @@ def convert(self, quantizer: QuantizerCls, node: Node, load_arg: Callable,
+                    # call_function
+                    scale, zero_point = activation_post_process.calculate_qparams()
+                    scale = float(scale)
+                    zero_point = int(zero_point)
+
+                    scale_arg, zero_point_arg = create_qparam_nodes(quantizer, node.name, scale, zero_point)
+
+                    assert not isinstance(node.target, str), "Expecting node.target for "
+                    "call_function to be a function instead of a string"
+                    quantized_op = get_quantized_operator(node.target)
+                    args = load_arg(quantized=[0])(node.args)
+                    kwargs = {**load_arg(quantized=False)(node.kwargs), "output_scale": scale_arg,
+                              "output_zero_point": zero_point_arg}
+                    if quantized_op in ARGS_TO_SKIP:
+                        args_to_skip = ARGS_TO_SKIP[quantized_op]
+                        for arg in args_to_skip:
+                            if arg in kwargs:
+                                kwargs.pop(arg)
+                    return quantizer.quantized_graph.create_node(
+                        "call_function", quantized_op, args, kwargs)
             else:
-                assert node.op == "call_function"
-                # call_function
-                scale, zero_point = activation_post_process.calculate_qparams()
-                scale = float(scale)
-                zero_point = int(zero_point)
-
-                scale_arg, zero_point_arg = create_qparam_nodes(quantizer, node.name, scale, zero_point)
-
-                assert not isinstance(node.target, str), "Expecting node.target for "
-                "call_function to be a function instead of a string"
-                quantized_op = get_quantized_operator(node.target)
-                args = load_arg(quantized=[0])(node.args)
-                kwargs = {**load_arg(quantized=False)(node.kwargs), "output_scale": scale_arg, "output_zero_point": zero_point_arg}
-                if quantized_op in ARGS_TO_SKIP:
-                    args_to_skip = ARGS_TO_SKIP[quantized_op]
-                    for arg in args_to_skip:
-                        if arg in kwargs:
-                            kwargs.pop(arg)
-                return quantizer.quantized_graph.create_node(
-                    "call_function", quantized_op, args, kwargs)
+                assert dtypes in [(torch.float16, torch.float16, None)]
+                # Generally fp16 kernels don't exist for fp16 ops
+                warnings.warn(
+                    "Only reference patterns are currently supported for {dtype} dtype with {op} op".format(dtype=dtypes, op=self.op))
+                return quantizer.quantized_graph.node_copy(node, load_arg(quantized=False))
         else:
-            assert dtypes == (torch.float16, torch.float16, None)
-            return quantizer.quantized_graph.node_copy(node, load_arg(quantized=False))
+            assert is_reference
```

Comment:
NIT: Why not check for this explicitly instead of having an else followed by an assert?

Reply:
I was mostly replicating lines like: https://github.com/pytorch/pytorch/blob/3802e577fbce9cb690d050d09b5288a3c97d5694/torch/quantization/fx/quantization_patterns.py#L225